### ISSUE TITLE
Conda list erroneously reports gdal as pypi package

### ISSUE THREAD

**Issue by scottyhq** [conda-forge/gdal-feedstock#636]
I've noticed strange behavior today trying to share a gdal environment with a colleage with `conda env export`. The problem is that conda seems to think gdal installed from conda-forge is actually from pypi. 

Older versions of gdal (<=3.3.2) from this feedstock behave as expected with `conda list` or `conda env export` showing gdal coming from conda-forge:

You can reproduce this with the following:
```
docker run --rm -it condaforge/mambaforge:4.12.0-0 /bin/bash
conda create -n gdal3.5 gdal=3.5
conda activate gdal3.5
conda list | grep gdal
# gdal                      3.5.0                    pypi_0    pypi
# libgdal                   3.5.0                h59d0e54_4    conda-forge
```
Note that `conda env export` will therefore create an environment.yml that puts gdal under the pip packages:
```
  - zlib=1.2.12=h166bdaf_1
  - zstd=1.5.2=h8a70e8d_1
  - pip:
    - gdal==3.5.0
prefix: /opt/conda/envs/gdal
```

This wasn't the case in the past, and sure enough after trying a few versions you'll see the `conda list` output is as expected:
```
conda create -n gdal3.3 gdal=3.3.2
conda activate gdal3.3
conda list | grep gdal
# gdal                      3.3.2            py39h218ed2d_4    conda-forge
# libgdal                   3.3.2                h9c9eb65_4    conda-forge
```

I also tried on macosx and see the same behavior.

**Comment by ocefpaf** [comment]
This is an old conda bug. I don't know if there is anything we can do in conda-forge. Pinging @jezdez and @chenghlee here for awareness.

**Comment by scottyhq** [comment]
@ocefpaf is there an existing issue in conda to link to?  I'm curious why gdal=3.3.2 works, is there some difference in metadata from that point forward?

**Comment by scottyhq** [comment]
Thanks for the insight @ocefpaf. For what it's worth, [`conda-lock`](https://github.com/conda-incubator/conda-lock) gets it right, so that is my current workaround: 
for an environment.yml such as
```
name: test
channels:
  - conda-forge
dependencies:
  - gdal=3.5
```

```
# conda-lock 1.0.5 from conda-forge
conda-lock lock -p linux-64
conda-lock render --kind env
```

**Comment by xylar** [comment]
I'm hopeful that #646 will fix this.  Let's do some tests in a couple of hours to see.

**Comment by zklaus** [comment]
This seems to be fixed now, at least with gdal-3.6.0. @scottyhq, is this still a problem for you?

**Comment by zklaus** [comment]
I think this is solved. If the problem resurfaces, please reopen.
